multiqc/modules/samblaster/samblaster.py

```diff
@@ -111,6 +117,9 @@ def parse_samblaster(self, f):
         if s_name is None:
             s_name = f["s_name"]
 
+        if version is not None:
+            self.add_software_version(version, s_name)
```

Comment:
The sample name could be parsed from the samblaster log file (if not reading from stdin). Thus we don't have the sample name until after the loop.

Reply:
Not sure I really understand. More verbose: Why can't we call the function as soon as `version` is assigned inside the loop (line 94), as we do in samtools?

Reply:
This is two examples to samblaster logs
```
==> testdata/data/modules/samblaster/samblaster_standalone.log <==
samblaster: Version 0.1.22
samblaster: Opening /home/vagrant/samblaster_standalone.sam for read.
samblaster: Outputting to stdout
samblaster: Loaded 86 header sequence entries.
samblaster: Marked 0 of 117 (0.00%) read ids as duplicates using 2752k memory in 0.000S CPU seconds and 0S wall time.

==> testdata/data/modules/samblaster/stdinout.txt <==
samblaster: Version 0.1.24
samblaster: Inputting from stdin
samblaster: Outputting to stdout
samblaster: Loaded 86 header sequence entries.
samblaster: Marked 32 of 2761 (1.16%) read ids as duplicates using 1600k memory in 0.004S CPU seconds and 18S wall time.
```

In both cases the version is parsed from the first line. In the first example the sample name will be parsed from the log file, see second line: `/home/vagrant/samblaster_standalone.sam`. In the second example the input is stdin so the sample name is taken from the file name (`stdinout.txt` in this case). As the sample name is needed when calling `add_software_version()` to avoid ignored samples we need to wait until we have parsed at least the second log file line. In practice it is just easier to do this after parsing the full file. Does this make sense?

Reply:
Gotcha, absolutely 👍🏻  Thanks for the great explanation 😃